src/lib/reasoners/shostak.ml

```diff
@@ -383,19 +383,33 @@ struct
       AC.is_mine_symb sb ty
     with
     | true  , false , false, false, false, false ->
+      Timers.with_timer Timers.M_Arith Timers.F_make @@ fun () ->
       ARITH.make t
+
     | false , true  , false, false, false, false ->
+      Timers.with_timer Timers.M_Records Timers.F_make @@ fun () ->
       RECORDS.make t
+
     | false , false , true  , false, false, false ->
+      Timers.with_timer Timers.M_Bitv Timers.F_make @@ fun () ->
       BITV.make t
+
     | false , false , false , true , false, false ->
+      Timers.with_timer Timers.M_Sum Timers.F_make @@ fun () ->
       ENUM.make t
+
     | false , false , false , false, true , false ->
+      Timers.with_timer Timers.M_Adt Timers.F_make @@ fun () ->
       ADT.make t
+
     | false , false , false , false, false, true  ->
+      Timers.with_timer Timers.M_AC Timers.F_make @@ fun () ->
       AC.make t
+
     | false , false , false , false, false, false ->
+      Timers.with_timer Timers.M_None Timers.F_make @@ fun () ->
```

Comment:
I don't think `M_None` should actually be used; let's keep `term_embed` untimed

Reply:
Or we can create a new module name: `M_Combine`.

Reply:
Sure, that works too

Reply:
Although `term_embed` is so trivial that I am not sure the timing overhead will not be more expensive than the function itself

Reply:
Right, for this one is useless.

Reply:
I removed it in this case.